### DISCUSSION TITLE
Size limit for each key / value

### DISCUSSION THREAD

**Discussion by Stevemoretz** [react-native-async-storage/async-storage#640] [General]
## You want to:  

Guys I need some help figuring out something about async storage.
on android it has a 6mb limit, I want to know what is its each key value pair limits for each key how big can be the value?

 ## Details:

If it uses a database I think the limit will be the database column max size limit :(

**Comment by krizzu** [comment]
For Android, SQLite is used, so [SQLite limits are appropriate](https://www.sqlite.org/limits.html). There's also an Android OS limitation on how big the data can be while reading from the DB (CursorWindow limitation) and that's around 2MB. I'm yet to find out why 6MB is a limit on Android, but I from what I've been testing, it can handle more than that (like normal SQLite). Maybe it was related to different storage used back when AsyncStorage used RockDB under the hood? You can increase this limit [by following instructions here](https://react-native-async-storage.github.io/async-storage/docs/advanced/db_size).

**Reply by Stevemoretz** [reply]
Hi thank you for the response, really appreciate it.

So that's the reading limit for each value right?

```
unlimited
The maximum length of TEXT is unlimited. SQLite supports various character encodings.
```
and for writing to each key there is no limits, because sqlite text field is unlimited.

Also thanks for that link, I was not asking about the whole 6mb limit, but just the value limit.
That 2mb limit helps a lot thanks for clarification.

**Reply by krizzu** [reply]
Because everything AsyncStorage saves in DB is string, we're limited by string/blob limitation of SQLite. The Android itself has limitation when reading values back from SQLite. I'll provide some warning when someone will try to save such big data.

**Reply by Stevemoretz** [reply]
I found on the Google that the text field of SQLite can have unlimited characters apparently is that true?
If so the 2mb limit only applies to the reading right?

What I mean is that we can even save more even 6mb in a key but we can't read in back, unless it is less than 2mb.

Am I right?
If so why not add this to the Readme
I think this is really important to know.

I could have messed my app up not knowing about the 2mb limit.
